### DISCUSSION TITLE
Keys storage

### DISCUSSION THREAD

**Discussion by gaetanoi** [Infineon/optiga-trust-m#78] [Q&A] [answered]
Hi,

How many keys (ECC and RSA) can I store at the same time? 
Is it possible to store four ecc key and two RSA key at the same time?

regards
Gaetano

**Comment by ayushev** [comment, marked as answer]
Hello @gaetanoi 

yes, at the same time you can store 4 ECC keys and 2 RSA Keys, but you have only 4 default certificate slots. There is however a space to store 3 Trust Anchors, these can be used as a certificate objects as well.

**Reply by gaetanoi** [reply]
Hi @ayushev, thanks for the reply.

crypto functions such as optiga_crypt_ecdsa_sign can use private key from a static key store or acquired session.
Is it possible to load a private key into the session context objects (0xE100 - 0xE103)?

**Reply by ayushev** [reply]
no, it is not possible. 
However, you can write a new private key into a key object (not session object) if a protected key update has been configured.

**Reply by gaetanoi** [reply]
do you mean generate a new private key? 
I want to use an externally provided private key.

**Reply by ayushev** [reply]
There is a procedure called Protected Key Update (Solution Reference Manual Section 6.7). It means that you can prepare a Key Object Settings (Metadata) so that it will be possible to upload your own key inside this object.
There are a couple of tools available which might help you to prepare required data.
